### PR TITLE
actionview: Make a block to url helpers optional

### PR DESCRIPTION
**Pull request by tk0miya**
These helper methods can be called without block.  So it would be better to declare as optional.

### CODE REVIEW

File: gems/actionview/6.0/actionview.rbs [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+      # You can also use custom data attributes using the <tt>:data</tt> option:
+      #
+      #   link_to "Visit Other Site", "http://www.rubyonrails.org/", data: { confirm: "Are you sure?" }
+      #   # => <a href="http://www.rubyonrails.org/" data-confirm="Are you sure?">Visit Other Site</a>
+      #
+      # Also you can set any link attributes such as <tt>target</tt>, <tt>rel</tt>, <tt>type</tt>:
+      #
+      #   link_to "External link", "http://www.rubyonrails.org/", target: "_blank", rel: "nofollow"
+      #   # => <a href="http://www.rubyonrails.org/" target="_blank" rel="nofollow">External link</a>
+      def link_to: (?untyped? name, ?untyped? options, ?untyped? html_options) ?{ () -> untyped } -> untyped
```
Comment (tk0miya) [inline gems/actionview/6.0/actionview.rbs:166]: Only what I did is adding a `?` before the block definition.

**Comment by ksss** [comment]
I would like `rbs diff` to have an option to read unified format...

Please wait a moment while I look at the diff.

**Comment by ksss** [comment]
@tk0miya Is this all the difference you intend?

```
$ bundle exec rbs diff --detail --format diff --type-name ActionView::Helpers::UrlHelper --before gems/actionview/6.0 --after tk0miya-actionview_urlhelper-patch
```

```diff
- [::ActionView::Helpers::UrlHelper public] def link_to: (?untyped? name, ?untyped? options, ?untyped? html_options) { () -> untyped } -> untyped
+ [::ActionView::Helpers::UrlHelper public] def link_to: (?untyped? name, ?untyped? options, ?untyped? html_options) ?{ () -> untyped } -> untyped

- [::ActionView::Helpers::UrlHelper public] def button_to: (?untyped? name, ?untyped? options, ?untyped? html_options) { () -> untyped } -> untyped
+ [::ActionView::Helpers::UrlHelper public] def button_to: (?untyped? name, ?untyped? options, ?untyped? html_options) ?{ () -> untyped } -> untyped

- [::ActionView::Helpers::UrlHelper public] def link_to_unless_current: (untyped name, ?::Hash[untyped, untyped] options, ?::Hash[untyped, untyped] html_options) { () -> untyped } -> untyped
+ [::ActionView::Helpers::UrlHelper public] def link_to_unless_current: (untyped name, ?::Hash[untyped, untyped] options, ?::Hash[untyped, untyped] html_options) ?{ () -> untyped } -> untyped

- [::ActionView::Helpers::UrlHelper public] def link_to_unless: (untyped condition, untyped name, ?::Hash[untyped, untyped] options, ?::Hash[untyped, untyped] html_options) { () -> untyped } -> untyped
+ [::ActionView::Helpers::UrlHelper public] def link_to_unless: (untyped condition, untyped name, ?::Hash[untyped, untyped] options, ?::Hash[untyped, untyped] html_options) ?{ () -> untyped } -> untyped

- [::ActionView::Helpers::UrlHelper public] def link_to_if: (untyped condition, untyped name, ?::Hash[untyped, untyped] options, ?::Hash[untyped, untyped] html_options) { () -> untyped } -> untyped
+ [::ActionView::Helpers::UrlHelper public] def link_to_if: (untyped condition, untyped name, ?::Hash[untyped, untyped] options, ?::Hash[untyped, untyped] html_options) ?{ () -> untyped } -> untyped

- [::ActionView::Helpers::UrlHelper public] def mail_to: (untyped email_address, ?untyped? name, ?::Hash[untyped, untyped] html_options) { () -> untyped } -> untyped
+ [::ActionView::Helpers::UrlHelper public] def mail_to: (untyped email_address, ?untyped? name, ?::Hash[untyped, untyped] html_options) ?{ () -> untyped } -> untyped
```

**Comment by tk0miya** [comment]
Yes. This commit https://github.com/ruby/gem_rbs_collection/pull/589/commits/6a6c44b1a9342534482d9f35d837e4ae99b970ed is what you want to see.

**Comment by tk0miya** [comment]
Now I updated this PR not to move the unrelated methods to actionview.rbs.

**Review by ksss — COMMENTED** [review]
APPROVE

**Pull request merged**